src/main/java/org/testeditor/fixture/web/WebDriverFixture.java

```diff
@@ -145,10 +154,89 @@ public WebDriver startBrowser(String browser) {
 		}
 		configureDriver();
 		return driver;
-    }
+	}
+
+	public void initWithReporter(TestRunReporter reporter) {
+		reporter.addListener(this);
+	}
+
+	@Override
+	public void reported(SemanticUnit unit, Action action, String msg) {
+		if (unit == SemanticUnit.TEST && action == Action.ENTER) {
+			runningTest = msg;
+		}
+		if (screenshotShouldBeMade(unit, action, msg)) {
+			screenshot(msg + '.' + action.name());
+		}
+	}
+
+	private String runningTest = null;
+	private static final int SCREENSHOT_FILENAME_MAXLEN = 128;
```

Comment:
The code ensures that each element within a path will be maximal SCREENSHOT_FILENAME_MAXLEN long. This should be ok with Windows 7. SWT-Implementation for screenshots uses the same code which works for windows, apparently.